tests/service/data/SyntaxTree/UnionCase/Recover Function Type 01.fs

```diff
@@ -0,0 +1,4 @@
+namespace Foo
+
+type Bar = | Bar of int -> int
```

Comment:
It looks like the the same parsing rules are used in `exception` types. Could you please also add a test for it?